### PR TITLE
0.1.0 [semver:minor]

### PR DESCRIPTION
**Pull request by gmemstr**
Lays the basic groundwork for working with APEX/LWC applications in the context of CircleCI.

This relies on https://github.com/CircleCI-Public/Salesforce-sfdx-cli-orb/pull/13 being merged _first_.

_updated semver pump to 0.1.0 for testing._

### CODE REVIEW

**Comment by gmemstr** [comment]
Apparently sfdx returns a 100 error during the unit testing, unsure why - digging a little deeper.

**Comment by gmemstr** [comment]
Turns out, it was a permissions issue. Resolved this by adding permissions and sample data importing to the `source-push` command.

**Review by KyleTryon — CHANGES_REQUESTED** [review]
Appears community data is not deploying. https://app.circleci.com/pipelines/github/CircleCI-Public/salesforce-apex-orb/29/workflows/6822751f-246b-468a-b434-7714e6d92134/jobs/76/parallel-runs/0/steps/0-110

Looks like sfdx force:mdapi:deploy:report will need to be ran, post meta-data deploy to actually trigger a fail if it does not work

**Comment by gmemstr** [comment]
Hm, good catch :(

**Comment by gmemstr** [comment]
@KyleTryon According to the test project, `-w 5` (wait 5 minutes) is the recommended approach, since it didn't fail but wasn't told to wait for completion. I think this would be easier than trying to check the status and adding another step. I've pushed a change to add this flag (with configuration), but curious what you think.

File: orb.yml [hunk 1]
```diff
@@ -0,0 +1,272 @@
+commands:
```
Comment (KyleTryon) [inline orb.yml]: orb.yml should not be in the source. There should have been an orb.yml entry in the .gitignore file, which may not have been generated if there was any issue using the orb starter kit.

File: src/@orb.yml [hunk 2]
```diff
@@ -3,10 +3,9 @@ version: 2.1
 description: >
   Test and deploy APEX applications to Salesforce.
 
-# This information will be displayed in the orb registry and is not mandatory.
 display:
-  home_url: "https://www.website.com/docs"
-  source_url: "https://www.github.com/EXAMPLE_ORG/EXAMPLE_PROJECT"
+  source_url: "https://github.com/CircleCI-Public/salesforce-apex-orb/"
```
Comment (KyleTryon) [inline src/@orb.yml]: Add the salesforce home url. I would recommend using https://developer.salesforce.com/

File: src/commands/mdapi-deploy.yml [hunk 3]
```diff
@@ -18,11 +18,16 @@ parameters:
     description: "Validate deploy, but don't save to org."
     type: boolean
     default: false
+  wait-time:
+    description: "Amount of time to wait for this to complete."
```
Comment (KyleTryon) [inline src/commands/mdapi-deploy.yml]: Let's replace "this" with a more formal "Maximum amount of time to complete deployment" or something similar.

**Review by KyleTryon — CHANGES_REQUESTED** [review]
How do you feel about first launching as a minor release as well to ensure everything is working before jumping to 1.0.0?

File: src/jobs/scratch-deploy-and-test.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 description: >
```
Comment (KyleTryon) [inline src/jobs/scratch-deploy-and-test.yml:1]: This job must support all of the parameters for each command within, including the two new wait time parameters.

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: orb.yml (hunk 1 above)
Comment (gmemstr) [reply orb.yml]: My bad, was using a local orb.yml file for pushing the alpha version for testing.

**Comment by gmemstr** [comment]
I agree with bumping to 0.1.0 before 1.0.0, updated title to reflect this :)

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,18 +1,109 @@
...
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circleci/salesforce-apex
+          context: orb-publishing
+          add-pr-comment: true
+          fail-if-semver-not-indicated: true
+          publish-version-tag: true
```
Comment (KyleTryon) [inline .circleci/config.yml:105]: Version tag is set to try but there are no SSH fingerprints.

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,18 +1,109 @@
...
+      # publish a semver version of the orb. relies on
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circleci/salesforce-apex
+          context: orb-publishing
+          add-pr-comment: true
```
Comment (KyleTryon) [inline .circleci/config.yml:101]: CPE bot needs to be listed at the bot user.

File: src/@orb.yml [hunk 7]
```diff
@@ -3,10 +3,9 @@ version: 2.1
 description: >
   Test and deploy APEX applications to Salesforce.
 
-# This information will be displayed in the orb registry and is not mandatory.
 display:
-  home_url: "https://www.website.com/docs"
-  source_url: "https://www.github.com/EXAMPLE_ORG/EXAMPLE_PROJECT"
+  source_url: "https://developer.salesforce.com"
```
Comment (KyleTryon) [inline src/@orb.yml]: source url should be this repo. 

This should be changed to home_url. add source_url

File: src/@orb.yml [hunk 8]
```diff
@@ -3,10 +3,9 @@ version: 2.1
...
 
 orbs:
-  sfdx-cli: circleci/salesforce-sfdx@1.0.0
\ No newline at end of file
+  # This is temporary while the sfdx Orb is being worked on.
+  sfdx: circleci/salesforce-sfdx@dev:alpha
```
Comment (KyleTryon) [inline src/@orb.yml]: When will this be updated?

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: src/@orb.yml (hunk 8 above)
Comment (gmemstr) [reply src/@orb.yml]: As soon as https://github.com/CircleCI-Public/Salesforce-sfdx-cli-orb/pull/13 is merged.

**Review by KyleTryon — APPROVED** [review]

**Pull request merged**